sky/utils/controller_utils.py

```diff
@@ -342,17 +342,19 @@ def replace_skypilot_config_path_in_file_mounts(
     # is provisioned, e.g., we may need to decide which cloud to create a bucket
     # to be mounted to the cluster based on the cloud the cluster is actually
     # launched on (after failover).
-    if file_mounts is None or not skypilot_config.loaded():
+    if file_mounts is None:
         return
     replaced = False
     with tempfile.NamedTemporaryFile('w', delete=False) as f:
-        new_skypilot_config = _setup_proxy_command_on_controller(cloud)
-        if new_skypilot_config is not None:
-            common_utils.dump_yaml(f.name, new_skypilot_config)
-            for remote_path, local_path in file_mounts.items():
-                if local_path == LOCAL_SKYPILOT_CONFIG_PATH_PLACEHOLDER:
-                    file_mounts[remote_path] = f.name
-                    replaced = True
+        if skypilot_config.loaded():
+            new_skypilot_config = _setup_proxy_command_on_controller(cloud)
+        else:
+            new_skypilot_config = {}
+        common_utils.dump_yaml(f.name, new_skypilot_config)
```

Comment:
Great catch! Can we remove the entry in the file mounts instead of creating an empty file to save the overhead for uploading the empty file?

Reply:
Done.